File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int max1=-1;
        int max2=-1;
        for(int i=0;i<arr.length;i++)
        {
            max1=Math.max(max1,arr[i]);
        }
        for(int i=0;i<arr.length;i++)
        {
            if(arr[i]!=max1)
            {
                max2=Math.max(max2,arr[i]);
            }
        }
        return max2;
    }
}

File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int max=arr[0],max2 = INT_MIN;;
        for(int i=1;i<arr.size();i++)
        {
           if(arr[i]>max)
           {
               max2=max;
               max=arr[i];
           }
           else if(arr[i]>max2 && max!=arr[i])
           {
               max2=arr[i];
           }
        }
        if(max2==INT_MIN)
        {
            return -1;
        }
        return max2;
    }
};